packages/flutter/lib/src/widgets/banner.dart

```diff
@@ -184,6 +198,18 @@ class Banner extends StatelessWidget {
   /// The message to show in the banner.
   final String message;
 
+  /// The directionality of the text.
+  ///
+  /// This is used to disambiguate how to render bidirectional text. For
+  /// example, if the message is an English phrase followed by a Hebrew phrase,
+  /// in a [TextDirection.ltr] context the English phrase will be on the left
+  /// and the Hebrew phrase to its right, while in a [TextDirection.rtl]
+  /// context, the English phrase will be on the right and the Hebrow phrase on
+  /// its left.
+  ///
+  /// Defaults to the ambient [Directionality], if any.
+  final TextDirection textDirection;
```

Comment:
Generally, I haven't been adding `textDirection` properties to everything that reads `Directionality.of` because you can always use the inherited widget to control things (and its rare to want to poke one widget rather than the whole subtree).

Reply:
I agree. I think Banner is a special case because it is reasonable for it to end up higher in the tree than the Directionality.